app/components/Views/MultichainAccounts/MultichainAccountConnect/MultichainAccountConnect.tsx

```diff
@@ -0,0 +1,796 @@
+          .build(),
+      );
+    },
+    [
+      accountsLength,
+      channelIdOrHostname,
+      trackEvent,
+      createEventBuilder,
+      eventSource,
+      hostInfo.metadata.isEip1193Request,
+      hostInfo.permissions,
+    ],
+  );
+
+  const navigateToUrlInEthPhishingModal = useCallback(
+    (url: string | null) => {
+      setShowPhishingModal(false);
+      cancelPermissionRequest(permissionRequestId);
+      navigation.goBack();
+      setIsLoading(false);
+
+      if (url !== null) {
+        navigation.navigate(Routes.BROWSER.HOME, {
+          screen: Routes.BROWSER.VIEW,
+          params: {
+            newTabUrl: url,
+            timestamp: Date.now(),
+          },
+        });
+      }
+    },
+    [cancelPermissionRequest, navigation, permissionRequestId],
+  );
+
+  const continueToPhishingSite = useCallback(() => {
+    setShowPhishingModal(false);
+  }, []);
+
+  const goToETHPhishingDetector = useCallback(() => {
+    navigateToUrlInEthPhishingModal(MM_PHISH_DETECT_URL);
+  }, [navigateToUrlInEthPhishingModal]);
+
+  const goToFilePhishingIssue = useCallback(() => {
+    navigateToUrlInEthPhishingModal(MM_BLOCKLIST_ISSUE_URL);
+  }, [navigateToUrlInEthPhishingModal]);
+
+  const goToEtherscam = useCallback(() => {
+    navigateToUrlInEthPhishingModal(MM_ETHERSCAN_URL);
+  }, [navigateToUrlInEthPhishingModal]);
+
+  const goBackToSafety = useCallback(() => {
+    navigateToUrlInEthPhishingModal(null); // No URL means just go back to safety without navigating to a new page
+  }, [navigateToUrlInEthPhishingModal]);
+
+  const triggerDappViewedEvent = useCallback(
+    (numberOfConnectedAccounts: number) =>
+      // Track dapp viewed event
+      trackDappViewedEvent({
+        hostname: hostnameFromUrlObj,
+        numberOfConnectedAccounts,
+      }),
+    [hostnameFromUrlObj],
+  );
+
+  const handleConnect = useCallback(async () => {
+    const request: PermissionsRequest = {
+      ...hostInfo,
+      metadata: {
+        ...hostInfo.metadata,
+        origin: channelIdOrHostname,
+      },
+      permissions: {
+        ...hostInfo.permissions,
+        ...getCaip25PermissionsResponse(
+          requestedCaip25CaveatValue,
+          selectedCaipAccountIds,
+          selectedChainIds,
+        ),
+      },
+    };
+
+    const connectedAccountLength = selectedAccountGroupIds.length;
+    const isMultichainRequest = !hostInfo.metadata.isEip1193Request;
+
+    try {
+      setIsLoading(true);
+      await Engine.context.PermissionController.acceptPermissionsRequest(
+        request,
+      );
+
+      triggerDappViewedEvent(connectedAccountLength);
+
+      trackEvent(
+        createEventBuilder(MetaMetricsEvents.CONNECT_REQUEST_COMPLETED)
+          .addProperties({
+            number_of_accounts: accountsLength,
+            number_of_accounts_connected: connectedAccountLength,
+            // TODO: Fix this. Not accurate
+            account_type: 'multichain', //getAddressAccountType(activeAddress),
```

Comment:
```suggestion
            account_type: 'multichain',
```

removing commented code